rfcs/text/0428-extending-types.md

```diff
@@ -0,0 +1,677 @@
+    },
+    "https://blockprotocol.org/@alice/property-type/age": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/age/v/1"
+    }
+  },
+  "required": [
+    "https://blockprotocol.org/@alice/property-type/name",
+    "https://blockprotocol.org/@alice/property-type/age"
+  ]
+}
+```
+
+and a _subtype_ `Employee`:
+
+```json
+{
+  "kind": "entityType",
+  "$id": "https://blockprotocol.org/@alice/entity-type/employee/v/2",
+  "type": "object",
+  "title": "Employee",
+  "allOf": [
+    { "$ref": "https://blockprotocol.org/@alice/entity-type/person/v/2" }
+  ],
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/name/v/1"
+    },
+    "https://blockprotocol.org/@alice/property-type/occupation": {
+      "$ref": "https://blockprotocol.org/@alice/property-type/occupation/v/1"
+    }
+  },
+  "required": [
+    "https://blockprotocol.org/@alice/property-type/name",
+    "https://blockprotocol.org/@alice/property-type/occupation"
+  ]
+}
+```
+
+The two types do share the `name` property, but the property definitions are compatible (even using the same version). It's possible to coerce an instance of `Employee` into an instance of `Person` as the properties compose and are compatible.
+For example, we may have the following `Employee` instance:
+
+```json
+{
+  "entityId": 112,
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": "Charles",
+    "https://blockprotocol.org/@alice/property-type/age": 35,
+    "https://blockprotocol.org/@alice/property-type/occupation": "Merchant"
+  }
+}
+```
+
+which can be coerced into the following `Person` instance:
+
+```json
+{
+  "entityId": 112,
+  "properties": {
+    "https://blockprotocol.org/@alice/property-type/name": "Charles",
+    "https://blockprotocol.org/@alice/property-type/age": 35
+  }
+}
+```
+
+## Subtyping
+
+We're already implicitly defining `{ "additionalProperties": false }` in the [versioning RFC](./0408-versioning-types.md#determining-type-compatibility) for all schemas, this RFC will piggyback on that, and change the existing, implicit `additionalProperties` usage to `unevaluatedProperties` and slightly change its [vocabulary definition](https://json-schema.org/draft/2020-12/json-schema-core.html#name-unevaluatedproperties).
+
+The vocabulary change is that the `unevaluatedProperties` keyword will only be applicable at top-level schemas (i.e. subtypes or schemas that don't specify `allOf`), and ignored when present on a schema that resides in the `allOf` array. Instead of the default value being `{ "unevaluatedProperties": {} }`, the default value will be `{ "unevaluatedProperties": false }`. Together this makes it so that all supertypes will be able to compose, as their individual, implicit `unevaluatedProperties` values won't have any effect on validation and defer evaluation to the extending type (the subtype).
```

Comment:
The top-level thing seems consistent with the intended usage, at least how it's described in issues like [this](https://github.com/ajv-validator/ajv/issues/1350#issuecomment-744656165)